pennylane/drawer/utils.py

```diff
@@ -159,48 +168,96 @@ def cwire_connections(layers, bit_map):
     ...     qml.cond(m0, qml.S)(3)
     >>> tape = qml.tape.QuantumScript.from_queue(q)
     >>> layers = drawable_layers(tape)
-    >>> bit_map, cwire_layers, cwire_wires = cwire_connections(layers)
-    >>> bit_map
-    {measure(wires=[0]): 0, measure(wires=[1]): 1}
+    >>> bit_map = {m0.measurements[0]: 0, m1.measurements[0]: 1}
+    >>> new_bit_map, cwire_layers, cwire_wires = cwire_connections(layers, bit_map)
+    >>> new_bit_map == bit_map # No reusage happening
+    True
     >>> cwire_layers
-    [[0, 2, 3], [1, 2]]
+    {0: [[0, 2, 3]], 1: [[1, 2]]}
     >>> cwire_wires
-    [[0, 0, 3], [1, 0]]
+    {0: [[0, 0, 3]], 1: [[1, 0]]}
 
     From this information, we can see that the first classical wire is active in layers
-    0, 2, and 3 while the second classical wire is active in layers 1 and 2.  The first "active"
+    0, 2, and 3 while the second classical wire is active in layers 1 and 2, with both classical
+    wires being used only once (the outer lists all have length 1). The first "active"
     layer will always be the one with the mid circuit measurement.
-
     """
     if len(bit_map) == 0:
-        return [], []
+        return bit_map, {}, {}
 
-    connected_layers = [[] for _ in bit_map]
-    connected_wires = [[] for _ in bit_map]
+    old_cwires = list(bit_map.values())
+    connected_layers = {cwire: [] for cwire in old_cwires}
+    connected_wires = {cwire: [] for cwire in old_cwires}
```

Comment:
New data structures: dictionaries with classical wires to lists. Later on, these lists will become lists of lists.
Note that indexing basically does not change, because we previously were using the `cwire` as integer indices into the list of lists above.